nats.go

```diff
@@ -1894,6 +1908,10 @@ func (nc *Conn) createConn() (err error) {
 
 // makeTLSConn will wrap an existing Conn using TLS
 func (nc *Conn) makeTLSConn() error {
+	if nc.Opts.SkipTLSWrapper {
+		// we do nothing when asked to skip the TLS wrapper
+		return nil
+	}
```

Comment:
I think we can achieve some of this with adding extra behavior to the implementation of the CustomDialer interface, to avoid having another option that is coupled to using a CustomDialer.  We can have an internal interface as follows in the client, which the CustomDialer could implement:

```go
type skipTLSDialer interface {
	SkipTLS() bool
}

// makeTLSConn will wrap an existing Conn using TLS.
func (nc *Conn) makeTLSConn() error {
	if nc.Opts.CustomDialer != nil {
		// we do nothing when asked to skip the TLS wrapper.
		sd, ok := nc.Opts.CustomDialer.(skipTLSDialer)
		if ok && sd.SkipTLS() {
			return nil
		}
	}
```

Then from the app perspective you would implement `Dial` and toggle the behavior by implementing `SkipTLS` like this:

```go
type testSkipTLSDialer struct {
	dialer *net.Dialer
}

func (sd *testSkipTLSDialer) Dial(network, address string) (net.Conn, error) {
	return sd.dialer.Dial(network, address)
}

func (sd *testSkipTLSDialer) SkipTLS() bool {
	return true
}
```

Reply:
I had the same though about relating that functionality to the custom dialer, but I failed to come up with a practical solution.  That said, I am happy with any way to have the skip functionality merged because I do not want to deal with a fork. This way seems very unobtrusive.

Reply:
Who can / will decide about the way of the implementation? What can I do?

Reply:
I have branch with the approach above on top of your commit, will send PR shortly for review

Reply:
opened PR with this change: https://github.com/nats-io/nats.go/pull/1147 /cc @piotrpio